camel-idea-plugin/src/main/java/org/apache/camel/idea/reference/endpoint/direct/DirectEndpointPsiElement.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.idea.reference.endpoint.direct;
+
+import java.util.Objects;
+import javax.swing.Icon;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.impl.RenameableFakePsiElement;
+import org.apache.camel.idea.reference.endpoint.CamelEndpoint;
+import org.apache.camel.idea.service.CamelPreferenceService;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class DirectEndpointPsiElement extends RenameableFakePsiElement {
+
+    private final CamelEndpoint endpoint;
+    private final PsiElement element;
+
+    public DirectEndpointPsiElement(@NotNull PsiElement element, @NotNull CamelEndpoint endpoint) {
+        super(null);
+        this.element = element;
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public String getName() {
+        return endpoint.getName();
+    }
+
+    @Override
+    public boolean isWritable() {
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public String getTypeName() {
+        return "direct endpoint";
+    }
+
+    @NotNull
+    @Override
+    public PsiElement getNavigationElement() {
+        return element.getNavigationElement();
+    }
+
+    @Nullable
+    @Override
+    public Icon getIcon() {
+        return CamelPreferenceService.getService().getCamelIcon();
+    }
+
+    @Override
+    public boolean isValid() {
+        return element.isValid();
+    }
+
+    @Override
+    public PsiElement getContext() {
+        return element.getNavigationElement();
+    }
+
+    @Override
+    public PsiFile getContainingFile() {
+        return element.getNavigationElement().getContainingFile();
+    }
+
+    @NotNull
+    @Override
+    public Project getProject() {
+        return element.getProject();
+    }
+
+    @Nullable
+    @Override
+    public String getLocationString() {
+        PsiFile file = getContainingFile();
+        if (file != null) {
+            return file.getName();
+        } else {
```

Comment:
No need for the else branch since you return early in the if branch

Just end the method with an `return null`